src/asynchronous/embassy/mod.rs

```diff
@@ -233,23 +234,39 @@ impl<'a, M: RawMutex, B: I2c> Tps6699x<'a, M, B> {
         self.controller.num_ports
     }
 
-    /// Wait for an interrupt to occur that satisfies the given predicate
+    /// Wait for an interrupt to occur that matches any bits in the given mask.
     pub async fn wait_interrupt(
         &mut self,
         clear_current: bool,
-        f: impl Fn(PortId, IntEventBus1) -> bool,
+        mask: [IntEventBus1; MAX_SUPPORTED_PORTS],
     ) -> [IntEventBus1; MAX_SUPPORTED_PORTS] {
         if clear_current {
             self.controller.interrupt_waker.reset();
         }
 
+        let mut acccumulated_flags = [IntEventBus1::new_zero(); MAX_SUPPORTED_PORTS];
         loop {
-            let flags = self.controller.interrupt_waker.wait().await;
-            for (port, flag) in flags.iter().enumerate() {
-                if f(PortId(port as u8), *flag) {
-                    return flags;
+            let mut done = false;
+            for ((port_index, &flags), mask) in
+                zip(self.controller.interrupt_waker.wait().await.iter().enumerate(), mask)
+            {
+                acccumulated_flags[port_index] |= flags;
+                let consumed_flags = flags & mask;
+                if consumed_flags != IntEventBus1::new_zero() {
```

Comment:
I think this should be checking all bits, not any bits
```suggestion
                if consumed_flags == mask {
```

Reply:
Any matches the current usages of the function better. Checking for command complete can be done by both any and all. The type-c service checks for any interrupt on any port: https://github.com/OpenDevicePartnership/embedded-services/blob/c9a635579034f9ef0638568f96dd1d8afc81f7e0/type-c-service/src/driver/tps6699x.rs#L241.

I renamed the function to `wait_interrupt_any` so we can implement `wait_interrupt_all` later if needed. I did also try keeping the predicate-based approach, but it ended up feeling kinda messy.